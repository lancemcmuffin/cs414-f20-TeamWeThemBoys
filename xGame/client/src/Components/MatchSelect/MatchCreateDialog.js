import { Button, TextField, Dialog, DialogContent, DialogContentText, DialogTitle } from '@material-ui/core';
import React, { useState } from "react";
import { Row, Col } from "reactstrap";
import PropTypes from "prop-types";
import { Formik } from "formik";
import { Select, MenuItem } from '@material-ui/core';
import { Alert } from "@material-ui/lab";

import * as userService from "../../service/userService";
import * as matchService from "../../service/matchService";


const MatchCreateDialog = (props) => {
    const {
        activeUser,
        showMatchCreation,
        setShowMatchCreation
    } = props;

    const [selectedOpponent, setSelectedOpponent] = useState("");
    const [searchError, setSearchError] = useState("");
    var [userSearchResults, setUserSearchResults] = useState([]);

    const handleSelectChange = (event) => {
        setSelectedOpponent(event.target.value);
    };

    const MakeItem = function(X) {
        return <MenuItem key={X.id} value={X.id}>{X.nickname}</MenuItem>;
    };

    return (
        <Dialog open={showMatchCreation} onClose={() => {setShowMatchCreation(false)}} aria-labelledby="form-dialog-title">
            <DialogTitle id="form-dialog-title">Match Creation</DialogTitle>
            <DialogContent>
                <DialogContentText>
                    To start a match, find an opponent to send a game request!
                </DialogContentText>
                <Row className="pb-3">
                  <Col>
                    {searchError && (
                      <Alert
                        onClose={() => {
                          setSearchError("");
                        }}
                        severity="error"
                      >
                        {searchError}
                      </Alert>
                    )}
                  </Col>
                </Row>
                <Formik
                    onSubmit={async (values) => {
                        if(values?.opponentValue) {
                            const user = await userService.search(values?.opponentValue)
                            if (user?.error) setSearchError(user?.message);
                            else setUserSearchResults({user});
                        }
                    }}
                    initialValues={{
                        opponentValue: "",
                    }}
                >
                    {(formProps) => {
                        const { setFieldValue, values } = formProps;
                        return (
                            <Col>
                                <Row className="pt-2">
                                    <Col xs="auto" className="d-flex justify-content-center">
                                        <TextField
                                            id="opponentValue"
                                            variant="outlined"
                                            //style={{ width: "80%" }}
                                            name="opponentValue"
                                            label="Search for a user"
                                            value={values?.["opponentValue"] || ""}
                                            onChange={(e) => setFieldValue("opponentValue", e.target.value)}
                                        />
                                    </Col>
                                    <Col xs="3" className="d-flex justify-content-center">
                                        <Button
                                            color="primary"
                                            variant="contained"
                                            style={{ textTransform: "none" }}
                                            onClick={formProps.handleSubmit}
                                        >
                                            Search
                                        </Button>
                                    </Col>
                                </Row>
                                <br />
                            </Col>
                        );
                    }}
                </Formik>
                <Formik
                    onSubmit={async (values) => {
                        const match = await matchService.createMatch(values);
                        if (match?.error) setSearchError(match?.message);
                        setShowMatchCreation(false);
                    }}
                    initialValues={{
                        blackID: "",
                        whiteID: activeUser?.id
                    }}
                >
                    {(formProps) => {
                        const { setFieldValue, values } = formProps;
                        return (
                            <Col>
                                <Row className="pt-2">
                                    <Col className="d-flex justify-content-center">
                                        <Select
                                            id="blackID"
                                            variant="outlined"
                                            style={{ width: "80%" }}
                                            name="blackID"
                                            onChange={handleSelectChange}
                                            value={values?.["blackID"] || ""}
                                            onChange={(e) => setFieldValue("blackID", e.target.value)}
                                        >
                                            {/*console.log("userSearchResults", userSearchResults)*/}
                                            {userSearchResults?.user?.map(MakeItem)}
                                        </Select>
                                    </Col>
                                </Row>
                                <br />
                                <Row className="pt-2">
                                    <Col className="d-flex justify-content-center">
                                        <Button
                                            color="primary"
                                            variant="contained"
                                            style={{ textTransform: "none" }}
                                            onClick={formProps.handleSubmit}
                                        >
                                            Create Match
                                        </Button>
                                    </Col>
                                    <Col className="d-flex justify-content-center">
                                        <Button
                                            size="small"
                                            color="primary"
                                            style={{ textTransform: "none" }}
                                            onClick={() => {setShowMatchCreation(false)}}
                                        >
                                            Cancel
                                        </Button>
                                    </Col>
                                </Row>
                                <br />
                            </Col>
                        );
                    }}
                </Formik>
            </DialogContent>
        </Dialog>
    );
};

MatchCreateDialog.propTypes = {
  setShowMatchCreation: PropTypes.func,
  activeUser: PropTypes.object,
  showMatchCreation: PropTypes.bool,
  userSearchResults: PropTypes.array,
};

MatchCreateDialog.defaultProps = {
  activeUser: {},
  setShowMatchCreation: () => {},
  showMatchCreation: false,
};

export default MatchCreateDialog;