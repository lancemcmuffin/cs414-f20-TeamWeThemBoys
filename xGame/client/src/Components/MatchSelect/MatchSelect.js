import React from "react";
import { Collapse } from "reactstrap";
import PropTypes from "prop-types";

const MatchSelect = (props) => {
  const { isOpen } = props;

  return (
    <Collapse isOpen={isOpen}>
      <div>
        <p>add match selection data</p>
      </div>
    </Collapse>
  );
};

MatchSelect.propTypes = {
  isOpen: PropTypes.bool
};

MatchSelect.defaultProps = {
  isOpen: true
};

export default MatchSelect;
