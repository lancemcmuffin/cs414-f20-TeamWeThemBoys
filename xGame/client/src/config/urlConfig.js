/**
 * @module urlConfig.js
 * @description exports URL constants used by services
 */

const localhostURL = "http://localhost:8080";
const serverURL = "tbd";

const serviceURL = localhostURL;

export const matchServiceURL = `${serviceURL}/match`;
export const userServiceURL = `${serviceURL}/user`;
export const messageServiceURL = `${serviceURL}/message`;
export const gameServiceURL = `${serviceURL}/game`;
