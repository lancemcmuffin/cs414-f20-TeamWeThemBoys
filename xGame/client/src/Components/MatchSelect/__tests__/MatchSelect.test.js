import "../../../enzyme.config.js";
import React from "react";
import { shallow } from "enzyme";

import MatchSelect from "../MatchSelect";

function testInitialState() {
  const app = shallow(<MatchSelect />);
  expect(app.html()).toMatchSnapshot();
}

test("Testing login initial state", testInitialState);
