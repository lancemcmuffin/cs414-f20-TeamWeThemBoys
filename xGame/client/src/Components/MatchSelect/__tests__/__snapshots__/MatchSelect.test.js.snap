// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing login initial state 1`] = `"<div class=\\"collapse show\\"><div><p>add match selection data</p></div></div>"`;
